packages/block-library/src/social-links/social-link/edit.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { URLInput } from '@wordpress/block-editor';
+import {
+	Button,
+	Dropdown,
+} from '@wordpress/components';
+/**
+ * Internal dependencies
+ */
+import socialList from '../social-list';
+import IconPicker from './icon-picker';
+
+// Receives an URL and tries to match to return an icon.
+const getIconByURL = ( url ) => {
+	const sites = Object.keys( socialList );
+	for ( const site of sites ) {
+		for ( const match of socialList[ site ] ) {
+			if ( url.includes( match ) ) {
+				return site;
+			}
+		}
+	}
+	return null;
+};
+
+const SocialLinkEdit = ( { attributes, setAttributes, isSelected } ) => {
+	const { icon, url } = attributes;
+
+	const classes = classNames( 'wp-social-icon', `wp-social-icon-${ icon }` );
```

Comment:
Is this something we want to maintain on WordPress core scope from now on? At the moment corresponding styles will be loaded only together with the block.

Reply:
I'm still not sure I understand this, this is how I would implement in a plugin, not quite sure how we should for core. Open to any suggestions